packages/flutter/test/material/text_field_test.dart

```diff
@@ -505,6 +508,52 @@ void main() {
     );
   }, skip: isBrowser);
 
+  testWidgets('text field toolbar options correctly changes options',
+      (WidgetTester tester) async {
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Scaffold(
+          body: Center(
+            child: Container(
+              width: 100,
+              height: 100,
+              child: const Opacity(
+                opacity: 0.5,
+                child: TextField(
+                  decoration: InputDecoration(hintText: 'Placeholder'),
+                  toolbarOptions: ToolbarOptions(copy: true),
+                ),
+              ),
+            ),
+          ),
+        ),
+      ),
+    );
+
+    await tester.showKeyboard(find.byType(TextField));
+
+    const String testValue = 'A B C';
+    tester.testTextInput
+        .updateEditingValue(const TextEditingValue(text: testValue));
+    await tester.pump();
+
+    // The selectWordsInRange with SelectionChangedCause.tap seems to be needed to show the toolbar.
```

Comment:
If you do it this way, then yes. selectWordsInRange will create the selection overlay which providing selection parameter to TextEditingValue alone will not.
you can look at this test for easier example https://github.com/flutter/flutter/blob/c4c7e033bd8bc4ea86885f8806625923ac1d3306/packages/flutter/test/material/text_field_test.dart#L1732

Reply:
SELECT ALL is is found in the widget tree because of selectable text, i am not seeing a way to test for it. But i does not show in the menu

Reply:
I thought select all is found due to this bug? https://github.com/flutter/flutter/pull/40713
or can you elaborate more how selectable text come into play in this test?

Reply:
No you are right, i tried a different approach now. what do you think?
I am looking for a number of widgets on the toolbar like other tests

Reply:
Since this is a material test, let's use material text selection. To get around the select all problem, you can purposely enable it because the real problem will be addressed in my pr.
See my example below
```
testWidgets('text field toolbar options correctly changes options',
          (WidgetTester tester) async {
    final TextEditingController controller = TextEditingController(
      text: 'Atwater Peel Sherbrooke Bonaventure',
    );
    await tester.pumpWidget(
      MaterialApp(
        home: Material(
          child: Center(
            child: TextField(
              controller: controller,
              toolbarOptions: const ToolbarOptions(
                copy: true,
                selectAll: true,
              ),
            ),
          ),
        ),
      ),
    );

    final Offset textfieldStart = tester.getTopLeft(find.byType(TextField));

    // This tap just puts the cursor somewhere different than where the double
    // tap will occur to test that the double tap moves the existing cursor first.
    await tester.tapAt(textfieldStart + const Offset(50.0, 5.0));
    await tester.pump(const Duration(milliseconds: 500));

    await tester.tapAt(textfieldStart + const Offset(150.0, 5.0));
    await tester.pump(const Duration(milliseconds: 50));
    // First tap moved the cursor.
    expect(
      controller.selection,
      const TextSelection.collapsed(
          offset: 8, affinity: TextAffinity.downstream),
    );
    await tester.tapAt(textfieldStart + const Offset(150.0, 5.0));
    await tester.pump();

    // Second tap selects the word around the cursor.
    expect(
      controller.selection,
      const TextSelection(baseOffset: 8, extentOffset: 12),
    );

    expect(find.text('COPY'), findsOneWidget);
    expect(find.text('SELECT ALL'), findsOneWidget);
    expect(find.text('PASTE'), findsNothing);
    expect(find.text('CUT'), findsNothing);
  }, skip: isBrowser);
```